### ISSUE TITLE
bug(Datepicker): NVDA keyboard interaction not working according to documentation

### ISSUE THREAD

**Issue by elenachiosa** [angular/components#27669]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

NVDA keyboard interaction for the calendar popup is not according to the [documentation](https://material.angular.io/components/datepicker/overview#keyboard-interaction).
JAWS works well.

### Reproduction

StackBlitz link: Can be reproduced on the Angular Material website: https://material.angular.io/components/datepicker/overview#datepicker-overview
Steps to reproduce:
1. Using the keyboard, open the datepicker popup
2. Press the right arrow
3. Press the down arrow

### Expected Behavior

When pressing the right arrow, the next day should be focused.
When pressing the down arrow, the same day in the row below (next week) should be focused.

### Actual Behavior

When pressing the right arrow, NVDA reads the current month letter by letter.
When pressing the down arrow, the next day is focused.

To get a bit more of a correct behavior, press Alt + arrows to get closer to the interaction described in the [documentation](https://material.angular.io/components/datepicker/overview#keyboard-interaction) (e.g. Alt  + Right Arrow focuses the next day). Sadly, Alt + Arrow Up closes the popup.

### Environment

- Angular: 16.2.0
- CDK/Material: 16.2.0
- Browser(s): Chrome
- Operating System (e.g. Windows, macOS, Ubuntu): Windows
- NVDA: 2023.1 (2023.1.0.27913)

**Comment by zarend** [comment]
Hello folks,

Thank you for reporting accessibility issues.

> NVDA keyboard interaction not working according to documentation

There's is no requirement that keyboard functionality is the same when using datepicker with Assistive Technology (Screen Reader). Screen Readers implement their own keyboard functionality and the specifics are usually an implementation detail of the screen reader.

Please let us know if you have an accessibility criteria, such as WCAG, or explanation why it is a problem for the user.

If you would like support, I suggest using the support chanels, which are [StackOverflow](https://stackoverflow.com/questions/tagged/angular-material2), [Gitter](https://gitter.im/angular/material2), and [Google Groups](https://groups.google.com/forum/#!forum/angular-material2). We can give a limited amount of support in this issue tracker.

Best Regards,

Zach
